planning.py

```diff
@@ -1446,30 +1446,33 @@ def find_hla(plan, hierarchy):
 	
     def making_progress(plan, initialPlan):
         """ 
-        Not correct
+        Prevents from infinite regression of refinements  
 
-        Normally should from infinite regression of refinements 
-        
-        Only case covered: when plan contains one action (then there is no regression to be done)  
+        (infinite regression of refinements happens when the algorithm finds a plan that 
+        its pessimistic reachable set intersects the goal inside a call to decompose on the same plan, in the same circumstances)  
         """
-        if (len(plan.action)==1):
-            return False
+        for i in range(len(initialPlan)):
+            if (plan == initialPlan[i]):
```

Comment:
This is better than the previous version, but I'm not sure if this is sufficient.